### DISCUSSION TITLE
Provisioning a feature pack in dev mode doesn't seem to work

### DISCUSSION THREAD

**Discussion by t1** [wildfly/wildfly-maven-plugin#335] [Q&A] [answered]
I really love to use the new `dev` mode. It's the first time I actually see a real benefit in using the `wildfly-maven-plugin`.

I'd like to add a feature pack (namely the [GraphQL feature pack](https://github.com/wildfly-extras/wildfly-graphql-feature-pack)), as it's suggested in the example in the [blog post announcing the dev mode](https://www.wildfly.org/news/2023/03/09/WildFly-Maven-Plugin/), but it's just not getting installed.

I took a brief look at the code, and it seems that the `dev` goal only provisions a default server; the `DevMojo` is a `AbstractServerStartMojo`, but maybe it would have to be a `AbstractProvisionServerMojo` or at least call it.

Or maybe the example is wrong, and I need to configure the `feature-pack-location` instead (where I could not install more than one feature pack?), but that fails with a `NullPointerException: Cannot invoke "org.jboss.galleon.universe.UniverseSpec.getFactory()" because "universeSpec" is null` in `UniverseFactoryLoader.java:109`, which I don't understand.

I have already opened [issue #111](https://github.com/wildfly-extras/wildfly-graphql-feature-pack/issues/111) in the GraphQL feature pack to add it to the quickstart example. There you can see my code.

**Comment by jamezp** [comment, marked as answer]
What should work is if you do `mvn clean wildfly:provision wildfly:dev`. That is not ideal I know and we should look at fixing this in the wildfly-maven-plugin. I thought I tested this, but I must have been testing wrong.

I've filed https://issues.redhat.com/browse/WFMP-206 for this as IMO it's a bug.
